### PR TITLE
Integrate game creation with datastore

### PR DESCRIPTION
**Pull request by ldcWV**
http://ldchen-step-2020.appspot.com/

### CODE REVIEW

File: src/main/java/com/google/sps/data/BasicGameData.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 * Represents only the info that is needed about a game from the main page.
 */
 public class BasicGameData {
-    MockDatastoreManager datastoreManager = new MockDatastoreManager();
+    private transient DatastoreManager datastoreManager = new DatastoreManager();
     //TODO(ldchen): delete all of the mocking once datastore is working.
```
Comment (saramissak) [inline src/main/java/com/google/sps/data/BasicGameData.java:26]: Remove this todo, because I believe you finished it.

Comment (saramissak) [reply src/main/java/com/google/sps/data/BasicGameData.java:26]: Ignore this comment if not.

File: src/main/java/com/google/sps/managers/DatastoreManager.java [hunk 2]
```diff
@@ -178,7 +178,7 @@ public Stage retrieveStage(String stageID) {
       return null;
```
Comment (saramissak) [inline src/main/java/com/google/sps/managers/DatastoreManager.java]: This isn't an issue but you can just do return; without putting null and it does the same thing.

Comment (ldcWV) [reply src/main/java/com/google/sps/managers/DatastoreManager.java]: Modified, thanks.

File: src/main/java/com/google/sps/managers/MockDatastoreManager.java [hunk 3]
```diff
@@ -178,4 +178,8 @@ private Game getRandomGame() {
         res.setTotalDifficulty(getRandomIntegerBetween(numDifficultyVotes, 3*numDifficultyVotes));
         return res.build();
     }
+
+    public boolean doesUsernameExist(String username) {
```
Comment (saramissak) [inline src/main/java/com/google/sps/managers/MockDatastoreManager.java:182]: Put a TODO so you dont forget or if I'm looking at ur code I know that this is still not implemented.

**Review by saramissak — APPROVED** [review]

File: src/main/java/com/google/sps/managers/MockDatastoreManager.java (hunk 3 above)
Comment (ldcWV) [reply src/main/java/com/google/sps/managers/MockDatastoreManager.java:182]: This is done. "return false" is all I need since this is just a mock class.

**Pull request merged**